src/main/java/org/eclipse/golo/compiler/GoloClassLoader.java

```diff
@@ -58,4 +62,25 @@ public GoloClassLoader() {
     }
     return lastClassIsModule;
   }
+  
+  /**
+   * Compiles and loads the resulting JVM bytecode for a Golo source file.
+   * This method declares a URL for the CodeSource of this class, which
+   * is useful for retrieving the location at runtime.
+   *
+   * @param goloSourceFilename    the source file name.
+   * @param sourceCodeInputStream the source input stream.
+   * @param sourceCodeLocation    the source file location.
+   * @return the class matching the Golo module defined in the source.
+   * @throws GoloCompilationException if either of the compilation phase failed.
+   */
+  public synchronized Class<?> load(String goloSourceFilename, InputStream sourceCodeInputStream, URL sourceCodeLocation) throws GoloCompilationException {
```

Comment:
This method is copy/past from the previous one. You should refactor the previous one to call this one with a `null` location.
Or even better, as stated in the main comments, refactor the method to only take a `CodeSource` and derive the name and input stream from it, and remove the old one.